cluster/certificates.go

```diff
@@ -0,0 +1,118 @@
+package cluster
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/pki"
+	"k8s.io/client-go/util/cert"
+)
+
+// SaveClusterCerts ...
+func SaveClusterCerts(crts map[string]pki.CertificatePKI) error {
+	logrus.Infof("[certificates] Save kubernetes certificates as secrets")
+	for crtName, crt := range crts {
+		err := saveCrtToKubernetes(crtName, crt)
+		if err != nil {
+			return fmt.Errorf("Failed to save certificate [%s] to kubernetes", crtName)
```

Comment:
We should return the exact error returned by saveCrtToKubernetes, along with the "Failed to save certificate" message